lib/src/components/NxTransferListHalf/__tests__/NxTransferListHalf.test.tsx

```diff
@@ -4,495 +4,537 @@
-          secondItem = list.find('.nx-transfer-list__item').at(1),
-          thirdItem = list.find('.nx-transfer-list__item').at(2),
-          firstItemMoveUpButton = firstItem.find('.nx-btn').at(0),
-          firstItemMoveDownButton = firstItem.find('.nx-btn').at(1),
-          secondItemMoveUpButton = secondItem.find('.nx-btn').at(0),
-          secondItemMoveDownButton = secondItem.find('.nx-btn').at(1),
-          thirdItemMoveUpButton = thirdItem.find('.nx-btn').at(0),
-          thirdItemMoveDownButton = thirdItem.find('.nx-btn').at(1);
-
-      expect(firstItem).toExist();
-      expect(firstItemMoveUpButton).toExist();
-      expect(firstItemMoveDownButton).toExist();
-      expect(firstItemMoveUpButton).toHaveClassName('disabled');
-      expect(firstItemMoveDownButton).not.toHaveClassName('disabled');
-
-      expect(secondItem).toExist();
-      expect(secondItemMoveUpButton).toExist();
-      expect(secondItemMoveDownButton).toExist();
-      expect(secondItemMoveUpButton).not.toHaveClassName('disabled');
-      expect(secondItemMoveDownButton).not.toHaveClassName('disabled');
-
-      expect(thirdItem).toExist();
-      expect(thirdItemMoveUpButton).toExist();
-      expect(thirdItemMoveDownButton).toExist();
-      expect(thirdItemMoveUpButton).not.toHaveClassName('disabled');
-      expect(thirdItemMoveDownButton).toHaveClassName('disabled');
+    beforeEach(function() {
+      // fixes "An update to ForwardRef(Tooltip) inside a test was not wrapped in act" errors
+      jest.useFakeTimers();
     });
-  });
-
-  it('passes onReorderItem with id and direction when move up or down button is clicked', function() {
-    const items = [{
-      id: 1,
-      displayName: 'a'
-    }, {
-      id: 2,
-      displayName: 'b'
-    }, {
-      id: 3,
-      displayName: 'c'
-    }];
-
-    const onReorderItem = jest.fn(),
-        component = getMounted({ allowReordering: true, items, onReorderItem }),
-        list = component.find('.nx-transfer-list__item-list');
-
-    expect(onReorderItem).not.toHaveBeenCalled();
 
-    const firstItem = list.find('.nx-transfer-list__item').at(0),
-        secondItem = list.find('.nx-transfer-list__item').at(1);
-    const moveDownButton = firstItem.find('.nx-btn').at(1);
-    const moveUpButton = secondItem.find('.nx-btn').at(0);
+    afterEach(runTimers);
 
-    expect(moveDownButton).toExist();
-    expect(moveUpButton).toExist();
+    async function runTimers() {
+      await act(async () => { await jest.runAllTimers(); });
+    }
 
-    moveDownButton.simulate('click');
+    async function advanceTimers(time: number) {
+      await act(async () => { await jest.advanceTimersByTime(time); });
+    }
 
-    expect(onReorderItem).toHaveBeenCalledWith(1, 1);
-
-    moveUpButton.simulate('click');
+    it('renders move up and down buttons inside .nx-transfer-list__item when allowReordering is true',
+        async function() {
+          jest.useRealTimers();
```

Comment:
I'm not sure this is the best resolution. On line 332, I want to check that the rendered buttons are indeed `Move Up` and `Move Down` buttons, and need to wait for the tooltip to initialize to confirm this. With `jest.useFakeTimers()` in the beforeEach, it causes this test to fail, and adding `await runTimers()` or `await advanceTimers(1000)` doesn't remedy the problem either. My understanding of jest's timers is still a bit shaky, so perhaps there is a better and cleaner way to resolve this issue than simply resetting the test to `jest.useRealTimers()`

Reply:
> adding await runTimers() or await advanceTimers(1000) doesn't remedy the problem either. 

I've wrestled quite a bit with the idiosyncracies of jest's fake timers, and this test is the most bizarre one yet.  If I modify what you have here to use an appropriately-placed `await runTimers()` instead of `jest.useRealTimers()`, it fails in the following scenarios:
* When I run `yarn test-watch` with this as the only locally-modified file (meaning that this file is the only one who's tests run)
* When I run `yarn test-watch` and press 'a' to toggle its "run all tests" option
* When I run `yarn jest src/components/NxTransferListHalf/__tests__/NxTransferListHalf.test.tsx`

However, it passes in every other scenario I try, including all of the following
* When I run `yarn test-watch` but with a few `console.log` statements added to `updateBatcher.ts`  These statements should have no meaningful effect on execution timing, but do cause a significant number of other tests to run besides the one in question
* When I run all tests with `yarn test`
* When I run just this test using `yarn jest --testNamePattern='renders move up and down buttons'`
* When I use `yarn test-watch`'s filter by test name feature with "renders move up and down buttons"

What I gather from all of this is that there must be some side effect that gets invoked simply from jest reading in one of the other files in our test suite (even if none of the tests in that file get executed), which causes this test to pass.  Only in execution modes where the other files aren't read at all does this test fail.  I will attempt to track down the side effect in question.

Reply:
> When I run yarn test-watch but with a few console.log statements added to updateBatcher.ts These statements should have no meaningful effect on execution timing, but do cause a significant number of other tests to run besides the one in question

Actually, for at least that scenario, I am now seeing it fail sometimes and not others.  Still investigating.

Reply:
> Actually, for at least that scenario, I am now seeing it fail sometimes and not others. Still investigating.

I think it's fixed with the changes in RSC-1336.  Please merge that branch into this one.  After doing so, you'll want to retarget this PR at that branch.  Not because it's actually going to get merged there, but so that the diff stays accurate

Reply:
> > Actually, for at least that scenario, I am now seeing it fail sometimes and not others. Still investigating.
> 
> I think it's fixed with the changes in RSC-1336. Please merge that branch into this one. After doing so, you'll want to retarget this PR at that branch. Not because it's actually going to get merged there, but so that the diff stays accurate

Merged with main here: [5eed414](https://github.com/sonatype/sonatype-react-shared-components/pull/880/commits/5eed41445b5e905b6c731b3ef177428f780c46e1) and removed redundant code here: [61e13d4](https://github.com/sonatype/sonatype-react-shared-components/pull/880/commits/61e13d48b2a6541ff75e5394972af6c54badf465). I checked to see if everything was passing after redundancies were removed, and all tests are passing.